baybe/acquisition/base.py

```diff
@@ -0,0 +1,71 @@
+"""Base classes for all acquisition functions."""
+
+from __future__ import annotations
+
+import warnings
+from abc import ABC
+from typing import ClassVar, Union
+
+from attrs import define
+
+from baybe.acquisition.adapter import debotorchize
+from baybe.serialization.core import (
+    converter,
+    get_base_structure_hook,
+    unstructure_base,
+)
+from baybe.serialization.mixin import SerialMixin
+from baybe.surrogates.base import Surrogate
+
+
+@define(frozen=True)
+class AcquisitionFunction(ABC, SerialMixin):
+    """Abstract base class for all acquisition functions."""
+
+    _abbreviation: ClassVar[str]
+    """An alternative name for type resolution."""
+
+    def to_botorch(self, surrogate: Surrogate, best_f: float):
```

Comment:
You assume here that `botorch` only needs the `surrogate` and `best_f` values. However, I do not think that this is true for all acqf functions that botorch offers, and it should be mentioned at some place that we currently only support these kind of acquisition functions for now.

Reply:
this is exactly what the next PR will deal with